sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1314,6 +1311,8 @@ static int mc_probe(struct platform_device *pdev)
 	if (!intel_ctx)
 		return -ENOMEM;
 
+	intel_ctx->platform_component = platform_component;
+
```

Comment:
I don't think this is needed at all, the path via asoc_sdw_init_dai_link() ignores the name and just sets it.

Reply:
I don't think there is a set on that path it passes the full platform component in, on the path through create_sdw_dailinks. Could you point to where the set happens?

Reply:
Since something is modifying it (the whole point of this PR) I just removed this saving and audio probed and worked fine ;)

Reply:
Hmm... that is very strange, it should get cleared on tear down but I don't think it should be getting set on creation. I guess I will go dig further, hopefully I can find some time.

Reply:
Ok yeah it seems it comes from soc_check_tplg_fes in soc-core, not really sure what that function does but it seems to go in and monkey around with all the platforms kinda makes me wonder if there is any point setting any of the platform stuff from the machine driver.